File: ScoutTrack/ScoutTrack.Services/Database/Entities/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ScoutTrack.Services.Database.Entities
{
    public class Category
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        public int MinAge { get; set; }

        [Required]
        public int MaxAge { get; set; }

        public string Description { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }

        public ICollection<Member> Members { get; set; } = new List<Member>();
    }
}

File: ScoutTrack/ScoutTrack.UI/scouttrack_desktop/lib/models/category.dart
import 'package:json_annotation/json_annotation.dart';

part 'category.g.dart';

@JsonSerializable()
class Category {
  final int id;
  final String name;
  final int minAge;
  final int maxAge;
  final String description;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Category({
    this.id = 0,
    this.name = '',
    this.minAge = 0,
    this.maxAge = 0,
    this.description = '',
    required this.createdAt,
    this.updatedAt,
  });

  factory Category.fromJson(Map<String, dynamic> json) =>
      _$CategoryFromJson(json);

  Map<String, dynamic> toJson() => _$CategoryToJson(this);
}
